core/common/src/Instant.kt

```diff
@@ -255,7 +266,23 @@ public fun Instant.minus(period: DateTimePeriod, timeZone: TimeZone): Instant =
  * @throws DateTimeArithmeticException if `this` or [other] instant is too large to fit in [LocalDateTime].
  *     Or (only on the JVM) if the number of months between the two dates exceeds an Int.
  */
-public expect fun Instant.periodUntil(other: Instant, timeZone: TimeZone): DateTimePeriod
+public fun Instant.periodUntil(other: Instant, timeZone: TimeZone): DateTimePeriod {
+    val thisLdt = toLocalDateTimeFailing(timeZone)
+    val otherLdt = other.toLocalDateTimeFailing(timeZone)
+    var thisDate = thisLdt.date
+    val otherDate = otherLdt.date
+    val months = thisDate.until(otherDate, DateTimeUnit.MONTH) // `until` on dates never fails
+    thisDate = thisDate.plus(months, DateTimeUnit.MONTH) // won't throw: thisLdt + months <= otherLdt, which is known to be valid
+    val days = thisDate.until(otherDate, DateTimeUnit.DAY) // `until` on dates never fails
+    if (days.absoluteValue > 31) {
+        throw DateTimeArithmeticException("The number of months between $this and $other does not fit in an Int")
+    }
+    thisDate = thisDate.plus(days, DateTimeUnit.DAY) // won't throw: thisLdt + days <= otherLdt
+    val thisInstant = LocalDateTime(thisDate, thisLdt.time).toInstant(timeZone)
```

Comment:
Does this `toInstant` use the same resolve mode as `ZDT.plus(days)` that was used before?

Reply:
Not exactly.

Imagine a situation:
* UTC offset before is `+1:00`;
* An overlap happens;
* UTC offset after is `+0:00`.

If we subtract, say, a day and land into this overlap, then, as is currently implemented, the `+0:00` offset will be used. In this PR, the UTC offset `+1:00` will be used instead.

Reply:
I believe it would be better to preserve resolve behavior in these operations, so that
 `Instant + DatePeriod(days = 1) == Instant.plus(1, DAY)`